packages/client/tests/functional/chunking-query/tests.ts

```diff
@@ -0,0 +1,263 @@
+      })
+
+      test('should succeed when the explicit number of params is 32767 or less', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsParams(ids)
+
+        expect(tags.length).toBe(n)
+      })
+    })
+
+    describe('chunking logic does not trigger with 2 IN filters, and results vary between Rust drivers and Driver Adapters', () => {
+      function selectWith2InFilters(ids: number[]) {
+        return prisma.tag.findMany({
+          where: {
+            OR: [
+              {
+                id: { in: ids },
+              },
+              {
+                id: { in: ids },
+              },
+            ],
+          },
+        })
+      }
+
+      describeIf(providerFlavor === undefined)('With Rust drivers only', () => {
+        // See: https://github.com/prisma/prisma/issues/21802.
+        test('Selecting 32767 ids at once in two inclusive disjunct filters results in error: "too many bind variables", but not with mysql', async () => {
+          const ids = generatedIds(32767)
+
+          try {
+            await selectWith2InFilters(ids)
+
+            if (!['mysql'].includes(provider)) {
+              // unreachable
+              expect(true).toBe(false)
+            }
+          } catch (error) {
+            const e = error as Error
+
+            if (['postgresql', 'cockroachdb'].includes(provider)) {
+              expect(e.message).toContain('Assertion violation on the database')
+              expect(e.message).toContain('too many bind variables in prepared statement')
+              expect(e.message).toContain(`expected maximum of 32767, received 65534`)
+            } else {
+              // unreachable
+              expect(true).toBe(false)
+            }
+          }
+        })
+
+        test('Selecting 32768 ids at once in two inclusive disjunct filters results in error: "too many bind variables"', async () => {
+          const ids = generatedIds(32768)
+
+          try {
+            await selectWith2InFilters(ids)
+            // unreachable
+            expect(true).toBe(false)
+          } catch (error) {
+            const e = error as Error
+
+            if (['postgresql', 'cockroachdb'].includes(provider)) {
+              expect(e.message).toContain('Assertion violation on the database')
+              expect(e.message).toContain('too many bind variables in prepared statement')
+              expect(e.message).toContain(`expected maximum of 32767, received 65535`)
+            } else {
+              expect(e.message).toContain('Prepared statement contains too many placeholders')
+            }
+          }
+        })
+      })
+
+      describeIf(providerFlavor !== undefined)('With Driver Adapters only', () => {
+        test('Selecting 32768 ids at once in two inclusive disjunct filters works', async () => {
+          const ids = generatedIds(32768)
+          await selectWith2InFilters(ids)
+        })
```

Comment:
It's missing an expect like 
```suggestion
        test('Selecting 32768 ids at once in two inclusive disjunct filters works', async () => {
          const ids = generatedIds(32768)
          await expect(selectWith2InFilters(ids)).resolves.toBeTruthy()
        })
```
`toBeTruthy()` is an example, maybe there is a better thing to expect here but I still need to read through things

Reply:
If that rejected, an error would be thrown, and the test would fail (so we'd see that in the CI).
Note: this is a straight-up copy of previously existing tests.